modules/programs/xplr.nix

```diff
@@ -0,0 +1,65 @@
+{ config, lib, pkgs, ... }:
+with builtins;
+let
+  inherit (lib)
+    types mkIf mkOption mkEnableOption mkPackageOptionMD mdDoc
+    literalExpression;
+
+  cfg = config.programs.xplr;
+  initialConfig = ''
+    version = '${cfg.package.version}'
+  '';
+  # we provide a default version line within the configuration file, which is obtained from the package's attributes
+  # merge the initial configFile, a mapped list of plugins and then the user defined configuration to obtain the final configuration
+  pluginPath = if cfg.plugins != [ ] then
+    (''
+      package.path=
+    '' + (concatStringsSep " ..\n" (map (p: ''"${p}/init.lua;"'') cfg.plugins))
+      + ''
+         ..
+        package.path
+      '')
+  else
+    "\n";
+  configFile = initialConfig + pluginPath + cfg.extraConfig;
+in {
+  meta.maintainers = [ maintainers.notashelf ];
+  options.programs.xplr = {
+    enable = mkEnableOption "xplr, terminal UI based file explorer";
+
+    package = mkPackageOptionMD pkgs "xplr" { };
```

Comment:
```suggestion
    package = mkPackageOptionMD pkgs "xplr" { nullable = true; };
```

Reply:
What for?

Reply:
Not every HM user uses nix to manage packages, and this allows setting to `null` directly instead of having to create a dummy package

Reply:
TBH this should even be set treewide, but no one's got around to it yet - see #3375